drivers/power/reset/tssupervisor-reset.c

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <linux/err.h>
+#include <linux/i2c.h>
+#include <linux/module.h>
+#include <linux/init.h>
+#include <linux/regmap.h>
+#include <linux/mfd/core.h>
+#include <linux/reboot.h>
+#include <linux/mfd/ts_supervisor.h>
+
+/* We need a static device to support this for shutdown/reboot hooks */
+static struct device *ts_rstc_device;
+static atomic_t ts_restart_nb_refcnt = ATOMIC_INIT(0);
+
+static ssize_t reboot_reason_show(struct device *dev,
```

Comment:
Ugh, I'm missing how this is getting called. Is there a macro that generates a reference to the symbol `reboot_reason_show`?

Reply:
After `reboot_reason_show()` is `ATTR_RO(reboot_reason)`; `ATTR_RO(name)` -> Generates a `struct device_attr` populated by `__ATTR_RO(name)` -> that has an entry referencing _name_`_show`. This is the preferred way to generate these tables/entries for `sysfs`.